docs/proposals/mapper-design.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Mapper Design
+status: Pending
+authors:
+    - "@sids-b"
+    - "@m1093782566"
+approvers:
+  - "@qizha"
+  - "@rohitsardesai83"
+creation-date: 2019-02-23
+last-updated: 2019-02-24
+---
+
+# Mapper Design
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non\-goals](#non-goals)
+  * [User cases](#user-cases)
+* [Proposal](#proposal)
+
+## Introduction
+Mapper is an application that that is used to connect and and control devices. Following are the responsibilities of mapper:
+1) Scan and connect to the device.
+2) Report the actual state of twin-attributes of device.
+3) Map the expected state of device-twin to actual state of device-twin.
+4) Collect telemetry data from device.
+5) Convert readings from device to format accepted by KubeEdge.
+6) Schedule events on the device.
+7) Check health of the device.
+
+Mapper can be specific to a protocol where standards are defined i.e Bluetooth, Zigbee, etc or specific to a device if it a custom protocol.
+
+## Motivation
+All devices can be connected and controller by drivers provided by drivers provided by their vendor.
+But the message from the device need to be translated into a format understood by KubeEdge.
+Also there should be a way to control the devices from the platform. Mapper is the application that interfaces between KubeEdge and devices.
+There should be a standard design for mappers supported by KubeEdge for keeping them generic and easy to use. 
+
+### Goals
+* A generic way to support multiple devices of different protocols by having a standard design for mappers provided by KubeEdge.
+* Easy to use Mapper design.
+* Open-up the possibility of having mapper SDK in future which can generate mapper based of configurable inputs.
+
+### Non-goals
+* Impose restriction on users to follow this design while writing applications for their device.
+* Have a single application that supports multiple devices of different protocols. 
+
+### User cases
+1) Manage expected/actual state of a device.
+2) Collect telemetry data from devices.
+3) Schedule events and check health of the device.
+
+## Proposal
+<img src="../images/proposals/mapper.png">
+
+Please find below the detailed explanation of each component of the mapper.
+
+**1) Event-Manager**: A device can be controlled by setting a specific value in physical register(s) of a device and readings can be acquired by getting the value from specific register(s).
```

Comment:
An event should be generated from the device. Any operations to a device should be defined as "action"

Reply:
Thanks @qizha , i have renamed to action :)